### PR TITLE
stm32: ethernet related changes

### PR DESCRIPTION
**Pull request by erwango**
Few ethernet changes for stm32:

*Add default configuration for ethernet support
*Provide minor updates to stm32 ethernet driver

### CODE REVIEW

File: drivers/ethernet/eth_stm32_hal.c [hunk 1]
```diff
@@ -340,8 +341,10 @@ static void eth0_iface_init(struct net_if *iface)
 
 	heth->Init.MACAddr = dev_data->mac_addr;
 
-	if (HAL_ETH_Init(heth) != HAL_OK) {
-		SYS_LOG_ERR("HAL_ETH_Init failed\n");
+	hal_ret = HAL_ETH_Init(heth);
+
+	if (hal_ret != HAL_OK) {
+		SYS_LOG_ERR("HAL_ETH_Init failed: %d\n", hal_ret);
 	}
```
Comment (galak) [inline drivers/ethernet/eth_stm32_hal.c]: Does it make sense to continue if HAL_ETH_Init isn't HAL_OK?

Comment (erwango) [reply drivers/ethernet/eth_stm32_hal.c]: Yes, sure, we can exit at this point, makes sense

File: drivers/ethernet/eth_stm32_hal.c [hunk 2]
```diff
@@ -340,8 +341,11 @@ static void eth0_iface_init(struct net_if *iface)
...
 	heth->Init.MACAddr = dev_data->mac_addr;
 
-	if (HAL_ETH_Init(heth) != HAL_OK) {
-		SYS_LOG_ERR("HAL_ETH_Init failed\n");
+	hal_ret = HAL_ETH_Init(heth);
+
+	if (hal_ret != HAL_OK) {
+		SYS_LOG_ERR("HAL_ETH_Init failed: %d\n", hal_ret);
+		return;
 	}
```
Comment (galak) [inline drivers/ethernet/eth_stm32_hal.c]: Does it make sense to move this before we call k_thread_create?

File: drivers/ethernet/eth_stm32_hal.c [hunk 3]
```diff
@@ -394,6 +403,8 @@ static struct eth_stm32_hal_dev_data eth0_data = {
 			.RxMode = ETH_RXINTERRUPT_MODE,
 			.ChecksumMode = ETH_CHECKSUM_BY_SOFTWARE,
 			.MediaInterface = ETH_MEDIA_INTERFACE_RMII,
+			.DuplexMode = ETH_MODE_FULLDUPLEX,
+			.Speed = ETH_SPEED_100M,
```
Comment (lpereira) [inline drivers/ethernet/eth_stm32_hal.c]: Will this disable auto negotiation of link speed and duplex mode?  Although we're in 2017, I can imagine this being plugged to a crusty switch that will only negotiate a half-duplex 10M link.

Comment (erwango) [reply drivers/ethernet/eth_stm32_hal.c]: You're right, I missed these are defined after auto negotiation. I'll remove this. Thanks!

File: drivers/ethernet/eth_stm32_hal.c (hunk 2 above)
Comment (erwango) [reply drivers/ethernet/eth_stm32_hal.c]: Yeah so we avoid having thread being created and left unused, this is right.

**Review by galak — APPROVED** [review]

**Pull request merged**